### PR TITLE
Arlo Baby: Support for night light brightness, color, temperature & rainbow effects

### PR DESCRIPTION
**Pull request by sebr**
Arlo Baby – adds support for the following features:
* Setting nightlight brightness
* Setting nightlight RGB value
* Enabling/disabling rainbow effect

Known issues:
* The entity correctly responds to changes in the nightlight color temperature however there is a limitation that the color temp cannot be set (nb; this could be worked around by converting a temperature to rgb value, the downside being it would not be shown correct in the Arlo app. Does this matter?)
* ~Brightness value does not get picked up on initialisation, only after the color is set (or brightness changed in the Arlo app). Help appreciated to figure this one out, I'm stumped.~

Cheers!

See also: #158 

Arlo Baby Light

### CODE REVIEW

**Comment by sebr** [comment]
@m14t are you interested in testing this?

**Comment by twrecked** [comment]
Again, thanks for this.

My one issue is I think you need to create a new brightness key specifically for the night light. The current brightness key is used by the camera and reusing this will cause problems. Also, if you make it generic enough - eg `LIGHT_BRIGHTNESS_KEY = lightBrightness` - I'll use it for the ArloLight too.

That might explain why your initial brightness is not working, the camera brightness will overwrite your nightlight brightness - or vice versa. If adding the key doesn't fix the initial brightness then:

Brightness might arrive a little differently in the initial resources packet than in the event packets you get from Arlo. If you haven't all ready, turn on packet dumps and restart HA. You are looking for the initial cameras settings and it arrives in a camera resources packet. Look for the following in dump:

```
 'resource': 'cameras',
```

This will be at the bottom of a packet of the current status of all the cameras you have. Inside is a `properties` array that lists the current settings of each camera. Somewhere in there should be the `nightLight` settings. This array of `properties` is split up and passed to the event handler of each camera, if this `nightLight` location is different than the event handing one you will need to add code to get this initial value.

**Comment by sebr** [comment]
Super, thanks for the pointers! I didn't realise the `BRIGHTNESS_KEY` is used by the camera for the image brightness, so I think you're spot on that the value is being clobbered.

File: custom_components/aarlo/pyaarlo/constant.py [hunk 1]
```diff
@@ -63,9 +65,10 @@
 
 # we can get these from the resource; doorbell is subset
 RESOURCE_KEYS = [ACTIVITY_STATE_KEY, AIR_QUALITY_KEY, AUDIO_DETECTED_KEY, BATTERY_KEY, BATTERY_TECH_KEY,
-                 BRIGHTNESS_KEY, CONNECTION_KEY, CHARGER_KEY, CHARGING_KEY, FLIP_KEY, HUMIDITY_KEY, LAMP_STATE_KEY,
-                 MIRROR_KEY, MOTION_DETECTED_KEY, MOTION_ENABLED_KEY,
-                 MOTION_SENS_KEY, POWER_SAVE_KEY, PRIVACY_KEY,
+                 BRIGHTNESS_KEY, CONNECTION_KEY, CHARGER_KEY, CHARGING_KEY, FLIP_KEY, HUMIDITY_KEY,
+                 LAMP_STATE_KEY, LIGHT_BRIGHTNESS_KEY, LIGHT_MODE_KEY,
```
Comment (sebr) [inline custom_components/aarlo/pyaarlo/constant.py]: @twrecked I hadn't added the `LIGHT_*` attributes to the `RESOURCE_KEYS` list so it wasn't being extracted from the initial event payload.

**Comment by twrecked** [comment]
I really like this, thanks for your hard work. I'm on my phone right now but I'll merge this request tomorrow morning. Then get a new release out shortly after.

**Comment by twrecked** [comment]
Thanks!

**Pull request merged**